### DISCUSSION TITLE
Allow mapping of containerlab ENV

### DISCUSSION THREAD

**Discussion by sdktr** [ipspace/netlab#1029] [Ideas]
I'd like to propose allowing environment variables to be mapped from the netlab topology.

https://containerlab.dev/manual/nodes/#env

The ENV can be set in quite some ways for containerlab. My simplistic view of this would be allowing env type 

env is already defined as a node property here:
https://github.com/ipspace/netlab/blob/a9688dc141adba233a1aa367de97c383b8bb5ad5/netsim/providers/clab.yml#L31

But not mapped in clab.j2: https://github.com/ipspace/netlab/blob/dev/netsim/templates/provider/clab/clab.j2

Related: we now start containerlab with sudo -E to preserve PATH env. Maybe this is not needed if we allow some other form of env mapping.

Related2: Clab docs: 

> You can also specify a magic ENV VAR - __IMPORT_ENVS: true - which will import all environment variables defined in your shell to the relevant topology level.

**Comment by ipspace** [comment]
Most of what you're looking for already exists -- node.clab.env is copied into node.env in clab.yml in https://github.com/ipspace/netlab/blob/dev/netsim/templates/provider/clab/clab.j2#L26.

I tested it with this topology:

```
provider: clab
defaults.device: eos
nodes:
  r1:
    clab.env:
      a: b
```

And this is the result:

```
name: X

mgmt:
  network: netlab_mgmt
  ipv4-subnet: 192.168.121.0/24
  # Note: 'start' not validated
topology:
  nodes:
    r1:
      mgmt-ipv4: 192.168.121.101
      kind: ceos
      env: {'INTFTYPE': 'et', 'CLAB_MGMT_VRF': 'management', 'a': 'b'}
      image: ceos:4.29.2F
      runtime: docker
```

Are you looking for something else?

**Reply by sdktr** [reply]
you're right, thanks for pointing this out, oversight on my part
